cmd/kubeadm/app/util/config/common.go

```diff
@@ -258,13 +259,17 @@ func MigrateOldConfig(oldConfig []byte, allowExperimental bool) ([]byte, error)
 		return []byte{}, err
 	}
 
+	gv := kubeadmapiv1old.SchemeGroupVersion
+	if allowExperimental {
+		gv = kubeadmapiv1.SchemeGroupVersion
+	}
```

Comment:
this should be covered by some testcases, pls see: https://github.com/kubernetes/kubernetes/pull/119075